### PR TITLE
Parametric ReLU

### PR DESCRIPTION
**Pull request by szagoruyko**
Parametric ReLU from this paper http://arxiv.org/abs/1502.01852
Has also test and doc with image.

### CODE REVIEW

File: generic/PReLU.c [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+      if(input->size[(input_ndim + 1) % 2] != nOutputPlane)
+	THError("wrong number of input planes");
+    }
+
+    real* output_data = THTensor_(data)(output);
+    real* input_data = THTensor_(data)(input);
+    real* weight_data = THTensor_(data)(weight);
+    long i,j,k;
+#pragma omp parallel for private(i)
```
Comment (nagadomi) [inline generic/PReLU.c]:
```C
#pragma omp parallel for private(j, k)
```

**Comment by nicholas-leonard** [comment]
Looks nice.

File: generic/PReLU.c (hunk 1 above)
Comment (szagoruyko) [inline generic/PReLU.c]: fixed thanks

**Comment by soumith** [comment]
looks good! squash your commits and I'll merge.
Mention in the doc that you should not apply weight decay to prelu units.

**Comment by soumith** [comment]
Ah i see you already have the "weight decay should not used on it" line.

**Comment by nicholas-leonard** [comment]
Also, could you add a link to the article to the doc?

**Comment by szagoruyko** [comment]
added the link to the doc and squashed

**Comment by soumith** [comment]
awesome! thanks sergey

**Pull request merged**